.github/workflows/checks.yaml

```diff
@@ -86,3 +102,50 @@ jobs:
         run: |
           echo "url=${{ needs.check.outputs.storybook_url }}">> $GITHUB_OUTPUT
           echo "Preview URL: ${{ needs.check.outputs.storybook_url}}"
+
+  analyze-bundle:
+    needs: check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.ref }}
+
+      - name: Hash Current Branch Name
+        id: hash_target_branch_name
+        run: |
+          hashed_name=$(printf "%s" "${{ github.base_ref }}" | base64)
+          echo "hashed=${hashed_name}">> $GITHUB_OUTPUT
+
+      - name: Download Current Branch Build Meta
+        uses: actions/download-artifact@v4
+        continue-on-error: true
+        with:
+          name: ${{steps.hash_curret_branch_name.outputs.hashed}}
+          path: 'current-branch'
+
+      - name: Set Up jq
+        run: sudo apt-get install -y jq
+
+      - name: List Artifacts
```

Comment:
Usually, we implement most of the logic using node scripts . you can move `list_artifacts`, `get_artifact_url` and `Download Target Branch Artifact` into `Analyze Bundles And Create Report` srcipt.